packages/http/server/response_builder.mare

```diff
@@ -0,0 +1,87 @@
+
+  // 4xx Client Error
+  :fun ref status_bad_request:            @_res << b"400 Bad Request\r\n", @
+  :fun ref status_unauthorized:           @_res << b"401 Unauthorized\r\n", @
+  :fun ref status_payment_required:       @_res << b"402 Payment Required\r\n", @
+  :fun ref status_forbidden:              @_res << b"403 Forbidden\r\n", @
+  :fun ref status_not_found:              @_res << b"404 Not Found\r\n", @
+  :fun ref status_method_not_allowed:     @_res << b"405 Method Not Allowed\r\n", @
+  :fun ref status_not_acceptable:         @_res << b"406 Not Acceptable\r\n", @
+  :fun ref status_proxy_authentication_required: @_res << b"407 Proxy Authentication Required\r\n", @
+  :fun ref status_request_timeout:        @_res << b"408 Request Time-out\r\n", @
+  :fun ref status_conflict:               @_res << b"409 Conflict\r\n", @
+  :fun ref status_gone:                   @_res << b"410 Gone\r\n", @
+  :fun ref status_length_required:        @_res << b"411 Length Required\r\n", @
+  :fun ref status_precondition_failed:    @_res << b"412 Precondition Failed\r\n", @
+  :fun ref status_entity_too_large:       @_res << b"413 Request Entity Too Large\r\n", @
+  :fun ref status_uri_too_large:          @_res << b"414 Request-URI Too Large\r\n", @
+  :fun ref status_unsupported_media_type: @_res << b"415 Unsupported Media Type\r\n", @
+  :fun ref status_range_not_satisfiable:  @_res << b"416 Requested range not satisfiable\r\n", @
+  :fun ref status_expectation_failed:     @_res << b"417 Expectation Failed\r\n", @
+
+  // 5xx Server Error
+  :fun ref status_internal_server_error: @_res << b"500 Internal Server Error\r\n", @
+  :fun ref status_not_implemented:       @_res << b"501 Not Implemented\r\n", @
+  :fun ref status_bad_gateway:           @_res << b"502 Bad Gateway\r\n", @
+  :fun ref status_service_unavailable:   @_res << b"503 Service Unavailable\r\n", @
+  :fun ref status_gateway_timeout:       @_res << b"504 Gateway Time-out\r\n", @
+  :fun ref status_version_not_supported: @_res << b"505 HTTP Version not supported\r\n", @
+
+  :fun ref header(name String'val, value String'val)
+    @_res << name.as_bytes << b": " << value.as_bytes << b"\r\n", @
+
+  :fun ref write(body String'val)
+    // Assumes headers section is already complete so it will start with an empty line.
+    @_res << b"\r\n"
+
+    @_res << body.as_bytes
+    @
+
+  // I think this next function is wrong per HTTP's definition when a body is present.
+  // We should only add the final empty line in cases where no message body is sent,
+  // like with the 204 No Content status, after all the headers.
+  //
+  // See https://stackoverflow.com/questions/13821263/should-newline-be-included-in-http-response-content-length
+  //
+  // This might be easier to deal with once we build the response from different
+  // fields instead of pilling everything into a single writable stream.
+  //
+  // Leaving the bug for now.
+  // TODO: fix above.
```

Comment:
@jemc Now that the specs use the `Bytes.join` function this bug is even more clear (because of the dual `b""` at the bottom) and I don't feel good merging it as is.

We need to support responses with no body where this new line must exist, but avoid it when a body was written. Since we can't inspect the response because it's a writable stream, what do you think if I flag the use of the `write` function (ie: `has_body = True`) and then use that information in `finish`?